src/mode/modeHandler.ts

```diff
@@ -220,9 +235,12 @@ export class SearchState {
         }
     }
 
-    constructor(direction: number, startPosition: Position, searchString = "") {
+    constructor(direction: number, startPosition: Position, vimSettings: VimSettings, searchString = "") {
         this.searchDirection = direction;
         this.searchCursorStartPosition = startPosition;
+        this.vimSettings = vimSettings;
+        // vimSettings must be put before assigining searchString 
+        // as we will run `_recalculateSearchRanges` immediately after the assignment.
```

Comment:
How do you feel about removing my setter for searchString, and instead use a method that takes both the search string and the settings object? It would eliminate this comment and bug.